common-methods.go

```diff
@@ -47,45 +51,78 @@ func isTargetURLDir(targetURL string) bool {
 	return true
 }
 
+// getSourcePartialFromAlias gets a ReadAtCloser from URL.
+func getSourcePartialFromAlias(alias string, urlStr string) (reader io.Reader, err *probe.Error) {
+	sourceClnt, err := newClientFromAlias(alias, urlStr)
+	if err != nil {
+		return nil, err.Trace(alias, urlStr)
+	}
+	reader, err = sourceClnt.Get()
+	if err != nil {
+		return nil, err.Trace(alias, urlStr)
+	}
+	return reader, nil
+}
+
 // getSource gets a reader from URL.
-func getSource(urlStr string) (reader io.ReadSeeker, err *probe.Error) {
+func getSourceStream(urlStr string) (reader io.Reader, err *probe.Error) {
 	alias, urlStrFull, _, err := expandAlias(urlStr)
 	if err != nil {
 		return nil, err.Trace(urlStr)
 	}
-	return getSourceFromAlias(alias, urlStrFull)
+	return getSourceStreamFromAlias(alias, urlStrFull)
 }
 
-// getSourceFromAlias gets a reader from URL.
-func getSourceFromAlias(alias string, urlStr string) (reader io.ReadSeeker, err *probe.Error) {
+// getSourceStreamFromAlias gets a reader from URL.
+func getSourceStreamFromAlias(alias string, urlStr string) (reader io.Reader, err *probe.Error) {
 	sourceClnt, err := newClientFromAlias(alias, urlStr)
 	if err != nil {
 		return nil, err.Trace(alias, urlStr)
 	}
-	return sourceClnt.Get(0, 0)
+	reader, err = sourceClnt.Get()
+	if err != nil {
+		return nil, err.Trace(alias, urlStr)
+	}
+	return reader, nil
+}
+
+// putTargetPartialFromAlias writes to URL from ReadAtCloser.
+func putTargetPartialFromAlias(alias string, urlStr string, reader io.Reader, size int64, progress io.Reader) (int64, *probe.Error) {
```

Comment:
This function can also be removed. putTargetStreamFromAlias can take progress reader. Caller can always pass nil when not needed.